code/code/sys/database.cc

```diff
@@ -9,347 +9,316 @@
 #include "toggle.h"
 
 namespace {
-    struct ltstr
-    {
-        bool operator()(const char* s1, const char* s2) const
-        {
-            return strcmp(s1, s2) < 0;
-        }
-    };
-
-
-    // we return this instead of null if they try to fetch an invalid column
-    const sstring empty="";
-
-    const char * db_connect[DB_MAX] = {
-        "sneezy",
-        "immortal",
-    };
-
-    // maintain instances of sneezydb and immodb
-    class TDatabaseConnection
-    {
-        MYSQL *databases[DB_MAX];
-        public:
-        TDatabaseConnection();
-        ~TDatabaseConnection();
-
-        const char *getConnectParam(dbTypeT type);
-        MYSQL *getDB(dbTypeT type);
-    };
-
-    TDatabaseConnection::TDatabaseConnection()
-    {
-        memset(databases, 0, sizeof(databases));
-    }
+  struct ltstr {
+      bool operator()(const char* s1, const char* s2) const {
+        return strcmp(s1, s2) < 0;
+      }
+  };
 
-    TDatabaseConnection::~TDatabaseConnection()
-    {
-        for (MYSQL* db : databases)
-          mysql_close(db);
-        mysql_thread_end();
-        mysql_library_end();
-    }
+  // we return this instead of null if they try to fetch an invalid column
+  const sstring empty = "";
 
-    TDatabaseConnection database_connection;
-}
+  const char* db_connect[DB_MAX] = {
+    "sneezy",
+    "immortal",
+  };
 
-std::vector <std::string> db_hosts(DB_MAX);
-std::vector <std::string> db_names(DB_MAX);
-std::vector <std::string> db_users(DB_MAX);
-std::vector <std::string> db_passwords(DB_MAX);
+  // maintain instances of sneezydb and immodb
+  class TDatabaseConnection {
+      MYSQL* databases[DB_MAX];
 
+    public:
+      TDatabaseConnection();
+      ~TDatabaseConnection();
 
-IDatabase::~IDatabase()
-{}
+      const char* getConnectParam(dbTypeT type);
+      MYSQL* getDB(dbTypeT type);
+  };
+
+  TDatabaseConnection::TDatabaseConnection() {
+    memset(databases, 0, sizeof(databases));
+  }
+
+  TDatabaseConnection::~TDatabaseConnection() {
+    for (MYSQL* db : databases) mysql_close(db);
+    mysql_thread_end();
+    mysql_library_end();
+  }
 
-const char *TDatabaseConnection::getConnectParam(dbTypeT type)
-{
+  TDatabaseConnection database_connection;
+}  // namespace
+
+std::vector<std::string> db_hosts(DB_MAX);
+std::vector<std::string> db_names(DB_MAX);
+std::vector<std::string> db_users(DB_MAX);
+std::vector<std::string> db_passwords(DB_MAX);
+
+IDatabase::~IDatabase() {}
+
+const char* TDatabaseConnection::getConnectParam(dbTypeT type) {
   assert(type >= 0);
   assert(type < DB_MAX);
-  if (db_names[type] != "")
-    return db_names[type].c_str();
+  if (db_names[type] != "") return db_names[type].c_str();
```

Comment:
Was it a considered decision to pull these on one line? I'm staring at it, at the moment, and it feels less readable than the 2-line version.

Reply:
That's something that's handled by a specific clang-format rule and is easily changed. I've gone back and forth on it myself, but remembered you saying you liked preserving vertical space, so I went with it. However, I have zero qualms about changing it as I don't have any strong feelings about it either way.

Were there any other stylistic things you didn't care for? Clang format has a ton of rules so most things can be changed and I'm pretty familiar with them at this point.

Reply:
There are ifs around that have so many parentheses and conditions and stuff that they don't really accommodate the next statement on the same line. I'd vote to keep them on the next line, unless they're trivial. Tho I'm afraid it's hard to teach clang-format what "trivial" means, so let's default to 2 lines.

Reply:
This rule is specifically configured to only put if statements that have no else clause on a single line. Any that do have an else clause, no matter how short, get put on separate lines. Knowing this, do you still have a preference? It doesn't make a difference to me, so whichever you want is fine.